### ISSUE TITLE
Button text overflows button borders at 200% text enlarge

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15359]
Button text overflows button borders at 200% text enlarge

-   **Severity:** Medium
-   **Affected Populations:**
    -   Low-Vision
    -   Cognitively Impaired
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Editor Top Bar
    -   Media Dialog

#### Issue description

Users who enlarge text alone to 200% will see the "Preview" button
misaligned with its borders in the Editor Top Bar. This may make reading
it more difficult for users with cognitive disabilities such as
dyslexia.

The ability to resize text is essential for users with low-vision, and
may be helpful for users who have a cognitive disability. Catering to
zoom alone is not sufficient because the browser's font-size may be
increased independently of zoom level.

##### Issue Code

```
    .components-button.is-button {


        ...


        line-height: 26px;


    }


    .components-button.is-large {


        ...


        line-height: 28px;


    }


    .edit-post-header .components-button.editor-post-preview {


        height: 33px;


        line-height: 32px;


    }





    .media-frame-title {


        position: absolute;


        top: 0;


        ...


        height: 50px;


        ...


    }


    .media-frame-router {


        position: absolute;


        top: 50px;


        ...


    }


    ...


    .wp-core-ui .button.button-large {


        height: 30px;


        line-height: 28px;


        padding: 0 12px 2px;


    }


    ...


    .wp-core-ui .button.button-hero {


        ...


        height: 46px;


        line-height: 44px;


        ...


    }

```

#### Remediation Guidance

Avoid setting fixed-pixel `height` for containers of text, since text
can expand separately from browser zoom.

Allow line-heights and container heights to grow naturally with text, by
using `min-height` and vertical `padding` instead of `height`.

##### Recommended Code

```
    .media-frame-title {


        ...


        min-height: 50px;


        ...


    }


    .media-frame-router {


        ...


    }


    ...


    .wp-core-ui .button.button-large {


        min-height: 30px;


    }


    ...


    .wp-core-ui .button.button-hero {


        ...


        min-height: 46px;


        ...


    }


```

#### Relevant standards

-   [1.4.4 Resize text (Level
    AA)](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-scale)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-2 in Tenon's report

**Comment by melchoyce** [comment]
Full report screenshot:

![image](https://user-images.githubusercontent.com/2846578/57186729-84e28480-6e99-11e9-8dc9-a1f5d840f445.png)

**Comment by getdave** [comment]
@karlgroves Can you confirm which browser this manifests in?

**Comment by afercia** [comment]
Discussed during today's accessibility bug-scrub, agreed there's not so much to add and just need to be fixed 🙂 

@getdave these zoom issues happen when zooming only the text. Can be reproduced with Firefox > Zoom > Only text. Or when altering the default font size of the operating system, as users with a11y needs do.

**Comment by kjellr** [comment]
This is closely related to #15356, which mentions other buttons that have this issue. The fix should be applied globally.

Note that the PR associated with that issue (#15973) fixes this right now. It may make sense to break that PR down to smaller pieces. Also, these buttons styles are reproduced from Core, so they should be fixed there as well.

**Comment by youknowriad** [comment]
Is 200% text-zoom the required zoom to test this? Where do we draw the line? Any guidelines about that?

**Comment by audrasjb** [comment]
@youknowriad here are the guidelines about 200% text zoom:
> Content satisfies the Success Criterion if it can be scaled up to 200%, that is, up to twice the width and height.
> The working group feels that 200% is a reasonable accommodation that can support a wide range of designs and layouts, and complements older screen magnifiers that provide a minimum magnification of 200%. Above 200%, zoom (which resizes text, images, and layout regions and creates a larger canvas that may require both horizontal and vertical scrolling) may be more effective than text resizing. Assistive technology dedicated to zoom support would usually be used in such a situation and may provide better accessibility than attempts by the author to support the user directly.
https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-scale.html

**Comment by youknowriad** [comment]
According to this description, I think this issue is fixed, here are the Gutenberg buttons with 200% text zoom (tested on firefox). There's still a breakage on the page but it's coming from the WP Admin bar.

<img width="504" alt="Capture d’écran 2020-02-17 à 9 14 25 AM" src="https://user-images.githubusercontent.com/272444/74635427-45ee9880-5166-11ea-80b3-bb02ecdbd95d.png">

<img width="1128" alt="Capture d’écran 2020-02-17 à 9 16 56 AM" src="https://user-images.githubusercontent.com/272444/74635434-48e98900-5166-11ea-8bef-d3536735711a.png">
